nibabel/streamlines/tractogram_file.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABCMeta, abstractmethod
+from nibabel.externals.six import with_metaclass
+
+from .header import Field
+
+
+class ExtensionWarning(Warning):
+    pass
+
+
+class HeaderWarning(Warning):
+    pass
+
+
+class HeaderError(Exception):
+    pass
+
+
+class DataError(Exception):
+    pass
+
+
+class abstractclassmethod(classmethod):
+    __isabstractmethod__ = True
+
+    def __init__(self, callable):
+        callable.__isabstractmethod__ = True
+        super(abstractclassmethod, self).__init__(callable)
+
+
+class TractogramFile(with_metaclass(ABCMeta)):
+    """ Convenience class to encapsulate tractogram file format. """
+
+    def __init__(self, tractogram, header=None):
+        self._tractogram = tractogram
+        self._header = {} if header is None else header
+
+    @property
+    def tractogram(self):
+        return self._tractogram
+
+    @property
+    def streamlines(self):
+        return self.tractogram.streamlines
+
+    @property
+    def header(self):
+        return self._header
+
+    @property
+    def affine(self):
+        return self.header.get(Field.VOXEL_TO_RASMM)
+
+    def get_tractogram(self):
```

Comment:
I'm fine with removing then.

Marc

On 16-06-06 09:06 PM, Matthew Brett wrote:
>
> In nibabel/streamlines/tractogram_file.py 
> <https://github.com/nipy/nibabel/pull/391#discussion_r65995954>:
>
> > +    def tractogram(self):
> > +        return self._tractogram
> > +
> > +    @property
> > +    def streamlines(self):
> > +        return self.tractogram.streamlines
> > +
> > +    @property
> > +    def header(self):
> > +        return self._header
> > +
> > +    @property
> > +    def affine(self):
> > +        return self.header.get(Field.VOXEL_TO_RASMM)
> > +
> > +    def get_tractogram(self):
>
> I personally prefer properties, as long as the conform to the nipy 
> property manifesto 
> <https://github.com/nipy/nibabel/wiki/property_manifesto> - and these 
> do. And I think it's a bad idea to have two ways to do the same thing. 
> I've removed these guys in my current PR to your branch.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nipy/nibabel/pull/391/files/302ed9abc0745c9882ae38a5b5af90a4b23b9eff#r65995954>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAoSJO4FBSJvmV9QuBsywnH2LXjE4lp5ks5qJMP9gaJpZM4GuLph>.
>